addons/mail/static/src/components/thread_needaction_preview/thread_needaction_preview.xml

```diff
@@ -39,9 +39,9 @@
                     </div>
                     <div class="o_ThreadNeedactionPreview_core">
                         <span class="o_ThreadNeedactionPreview_coreItem o_ThreadNeedactionPreview_inlineText" t-att-class="{ 'o-empty': inlineLastNeedactionMessageBody.length === 0 }">
-                            <t t-if="thread.lastMessage and thread.lastMessage.author">
+                            <t t-if="thread.lastNonTransientMessage and thread.lastNonTransientMessage.author">
                                 <MessageAuthorPrefix
-                                    messageLocalId="thread.lastMessage.localId"
+                                    messageLocalId="thread.lastNonTransientMessage.localId"
```

Comment:
Wondering but shouldn't those actually be `lastNeedactionMessage` instead? cc @alexkuhn 

It's strange that in `o_ThreadNeedactionPreview_header` we use that field for the date, but then we display the last  message and not last needaction message which corresponded to the date.

Reply:
Indeed, should be `lastNeedactionMessage` here.

Reply:
Then what would have to be tested in the scope of this PR for needaction's ?

Reply:
Not sure to understand to question, but if it's how to reproduce the issue of not using `lastNeedactionMessage` here, it's when there's a needaction message that is not the last message of a conversation. You should see that author prefix uses author of last message instead of author of last needaction message. There's a mismatch with the inline body, which is on the last needaction message.

Reply:
Test we see the proper last needaction message info everywhere, even if there is a more recent (not needaction) message.

Reply:
Just like any other test, a hint that it covers the issue is that test should be red if you revert the change.